core/src/main/java/com/scalar/db/api/OperationBuilder.java

```diff
@@ -305,6 +305,30 @@ interface ImplicitPreReadEnabled<T> {
     T implicitPreReadEnabled(boolean implicitPreReadEnabled);
   }
 
+  interface InsertModeEnabled<T> {
```

Comment:
It might be better to have `PutMode` enum (or something) and have `INSERT`, `UPSERT`, and `UPDATE` to match the corresponding SQL commands?
Not fully sure, but it might be worth thinking now.

Reply:
Let me explain the background. We plan to introduce `insert()`, `upsert()`, and `update()` methods to the transaction API (`TransactionCrudOperable`). Additionally, we intend to mark the `put()` method in the transaction API as deprecated. The insert mode will be used in the `insert()` method.

What do you think?